code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp

```diff
@@ -0,0 +1,283 @@
+
+namespace lmvp {
+
+using odcore::base::module::DataTriggeredConferenceClientModule;
+using odcore::base::Lock;
+using odcore::data::Container;
+using odcore::data::image::SharedImage;
+using odcore::exceptions::ValueForKeyNotFoundException;
+using odcore::wrapper::SharedMemoryFactory;
+using odcore::wrapper::SharedMemory;
+
+void onMouse(int event, int x, int y, int flags, void * userdata);
+
+void onMouse(int event, int x, int y, int flags, void * userdata) {
+    LMVP_UNUSED(flags);
+    LMVP_UNUSED(userdata);
+
+    if(event == cv::EVENT_MOUSEMOVE) {
+        std::cout << "x: " << x << " y: " << y << std::endl;
+    }
+}
+
+class VanishingPointLowPass {
+public:
+    VanishingPointLowPass(float dampingFactor) : dampingFactor_(dampingFactor), lastVanishingPoint_() { }
+
+    std::shared_ptr<cv::Point2f> lowpass(std::shared_ptr<cv::Point2f> vanishingPoint) {
+        if(!vanishingPoint) {
+            //lastVanishingPoint_ = vanishingPoint;
+            return vanishingPoint;
+        }
+
+        if(!lastVanishingPoint_) {
+            lastVanishingPoint_ = vanishingPoint;
+            return vanishingPoint;
+        }
+
+        cv::Point2f distance = *vanishingPoint - *lastVanishingPoint_;
+
+        lastVanishingPoint_ = std::make_shared<cv::Point2f>(dampingFactor_ * distance + *lastVanishingPoint_);
+
+        return lastVanishingPoint_;
+    }
+
+private:
+    const float dampingFactor_;
+
+    std::shared_ptr<cv::Point2f> lastVanishingPoint_;
+};
+
+class FailedAttachingToSharedMemoryException : public std::runtime_error {
+public:
+    FailedAttachingToSharedMemoryException(const std::string & name)
+        : std::runtime_error("Failed attaching to shared memory " + name) {
+    }
+};
+
+/**
+ * Class linking a SharedImage to a cv::Mat. Attachment to the shared memory happens on construction.
+ * Construction only succeeds when attaching to the shared memory was successful.
+ */
+class SharedImageCvAdapter {
+public:
+    SharedImageCvAdapter(const SharedImage & sharedImage) : sharedImageMemory_(), imageHeader_() {
+        attachToSharedMemory(sharedImage);
+    }
+
+    cv::Mat fetchImage() {
+        Lock lock(sharedImageMemory_);
+
+        return imageHeader_.clone();
+    }
+
+private:
+    void attachToSharedMemory(const SharedImage & sharedImage) {
+        sharedImageMemory_ = SharedMemoryFactory::attachToSharedMemory(sharedImage.getName());
+
+        if(!sharedImageMemory_->isValid()) {
+            throw FailedAttachingToSharedMemoryException(sharedImage.getName());
+        }
+
+        imageHeader_ = cv::Mat(sharedImage.getHeight(), sharedImage.getWidth(),
+                               CV_MAKETYPE(CV_8U, sharedImage.getBytesPerPixel()),
+                               sharedImageMemory_->getSharedMemory());
+    }
+
+    std::shared_ptr<SharedMemory> sharedImageMemory_;
+    cv::Mat imageHeader_;
+};
+
+class LaneMarkingsVanishingPointDetectionModule : public DataTriggeredConferenceClientModule {
```

Comment:
Please separate the classes into a header and implementation file (separate header and implementation files for the individual classes).